docs/sql-ref-ansi-compliance.md

```diff
@@ -165,6 +165,7 @@ The behavior of some SQL operators can be different under ANSI mode (`spark.sql.
   - `map_col[key]`: This operator throws `NoSuchElementException` if key does not exist in map.
   - `CAST(string_col AS TIMESTAMP)`: This operator should fail with an exception if the input string can't be parsed.
   - `CAST(string_col AS DATE)`: This operator should fail with an exception if the input string can't be parsed.
+  - `CAST(string_col AS BOOLEAN)`: This operator should fail with an exception if the input string can't be parsed.
```

Comment:
not related to this PR. Does section `### Type Conversion` cover the CAST behaviors?

Reply:
Yes and no.
```
When `spark.sql.ansi.enabled` is set to `true`, explicit casting by `CAST` syntax throws a runtime exception for illegal cast patterns defined in the standard, e.g. casts from a string to an integer.
```
covers this, but we had better write down all the cases that would throw exceptions.

How about a follow-up to add a subsection for all these cases under `### Type Conversion`?

Reply:
SGTM